stream-chat-android-core/src/main/java/io/getstream/chat/android/models/streamcdn/image/extensions/StringExtensions.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.models.streamcdn.image.extensions
+
+import androidx.annotation.FloatRange
+import io.getstream.chat.android.models.streamcdn.image.StreamCdnCropImageMode
+import io.getstream.chat.android.models.streamcdn.image.StreamCdnOriginalImageDimensions
+import io.getstream.chat.android.models.streamcdn.image.StreamCdnResizeImageMode
+import io.getstream.log.StreamLog
+
+/**
+ * Returns [StreamCdnOriginalImageDimensions] if the image is hosted by Stream's CDN and is resizable,
+ * otherwise returns null.
+ *
+ * @return Class containing the original width and height dimensions of the image or null.
+ */
+public fun String.getStreamCdnHostedImageDimensions(): StreamCdnOriginalImageDimensions? {
+    return try {
+        val width = this.substringAfter("ow=", "").takeIf { it.isNotBlank() }
+            ?.substringBefore("&")?.toInt()
```

Comment:
Maybe use `toIntOrNull()` to safeguard against bad formatting? Then we might be able to avoid the try/catch block?

Reply:
The try catch block is here on purpose in order to not obscure any parsing errors.
If we used `toIntOrNull()` we wouldn't be able to log any errors and would just fail silently without any way of knowing what happened.